### PR TITLE
Refactor TCK Runner and ensure TCK result files are well formed

### PR DESCRIPTION
**Pull request by KyleAure**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Introduce changes to the TCK Runner to make it more developer friendly

- Make TCKRunner a builder class (easier future updates without the need to introduce a new overloaded constructor)
- Add platformVersion and qualifier fields to TCKRunner for better Jakarta support
- Remove unused fields from TCKRunner
- Move result generation logic from TCKResultWriter to TCKResultInfo class where it is better suited for unit testing
- Add unit testing for TCKResultInfo
- Rollout refactor builder to FAT tests

### CODE REVIEW

**Comment by KyleAure** [comment]
#build - run personal build as proof of concept

**Comment by KyleAure** [comment]
#build - personal build
#libby - check headers

File: dev/com.ibm.ws.microprofile.graphql_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java [hunk 1]
```diff
@@ -56,12 +56,10 @@ public static void tearDown() throws Exception {
...
-    public void testGraphQL10Tck() throws Exception {
-        String bucketName = "com.ibm.ws.microprofile.graphql_fat_tck";
-        String testName = this.getClass() + ":testGraphQL10Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "GraphQL";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+    public void testGraphQL10Tck() throws Exception {        
+        TCKRunner.build(server, Type.MICROPROFILE, "GraphQL")
+        	.withDefaultSuiteFileName()
+        	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.graphql_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.openapi.1.1_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java [hunk 2]
```diff
@@ -58,11 +58,10 @@ public void testOpenAPI11Tck() throws Exception {
...
 
-        String bucketName = "com.ibm.ws.microprofile.openapi.1.1_fat_tck";
-        String testName = this.getClass() + ":testOpenAPI11Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open API";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, additionalProps);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open API")
+    		.withDefaultSuiteFileName()
+    		.withAdditionalMvnProps(additionalProps)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.openapi.1.1_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java [hunk 3]
```diff
@@ -57,12 +57,11 @@ public void testOpenAPI10Tck() throws Exception {
...
-        String bucketName = "com.ibm.ws.microprofile.openapi_fat_tck";
-        String testName = this.getClass() + ":testOpenAPI10Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open API";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, additionalProps);
+        
+        TCKRunner.build(server, Type.MICROPROFILE, "Open API")
+        	.withDefaultSuiteFileName()
+        	.withAdditionalMvnProps(additionalProps)
+        	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.openapi_fat_tck/fat/src/org/eclipse/microprofile/openapi/fat/tck/OpenAPITckTest.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.opentracing.1.1_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing11/tck/OpentracingTCKLauncher.java [hunk 4]
```diff
@@ -66,10 +58,8 @@ public static void tearDown() throws Exception {
...
     @AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
     public void launchOpenTracing11Tck() throws Exception {
-    	String bucketName = "com.ibm.ws.opentracing.1.1_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing11Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withDefaultSuiteFileName()
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing.1.1_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing11/tck/OpentracingTCKLauncher.java]: Tab-space mix (in fairness this one had a tab in it already)

File: dev/com.ibm.ws.microprofile.opentracing.1.2_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing12/tck/OpentracingTCKLauncher.java [hunk 5]
```diff
@@ -49,10 +49,8 @@ public static void tearDown() throws Exception {
...
     @AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
     public void launchOpenTracing12Tck() throws Exception {
-        String bucketName = "com.ibm.ws.opentracing.1.2_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing12Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withDefaultSuiteFileName()
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing.1.2_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing12/tck/OpentracingTCKLauncher.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingRestClientTCKLauncher.java [hunk 6]
```diff
@@ -59,10 +59,8 @@ public static void tearDown() throws Exception {
...
     public void launchOpenTracing13RestClientTck() throws Exception {
         String suiteName = "rest-client-tck-suite.xml";
-        String bucketName = "com.ibm.ws.opentracing.1.3_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing13RestClientTck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withSuiteFileName(suiteName)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingRestClientTCKLauncher.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingTCKLauncher.java [hunk 7]
```diff
@@ -49,12 +49,8 @@ public static void tearDown() throws Exception {
...
-        // Use default tck-suite.xml
-        
-        String bucketName = "com.ibm.ws.opentracing.1.3_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing13Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withDefaultSuiteFileName()
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingTCKLauncher.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingTCKLauncherMicroProfile.java [hunk 8]
```diff
@@ -50,10 +50,8 @@ public static void tearDown() throws Exception {
...
     public void launchOpenTracing13RestClientTck() throws Exception {
         String suiteName = "tck-and-rest-client-tck.xml";
-        String bucketName = "com.ibm.ws.opentracing.1.3_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing13RestClientTck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withSuiteFileName(suiteName)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing.1.3_fat_tck/fat/src/com/ibm/ws/microprofile/opentracing13/tck/OpentracingTCKLauncherMicroProfile.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.opentracing_fat_tck/fat/src/org/eclipse/microprofile/opentracing/fat/tck/OpentracingTCKLauncher.java [hunk 9]
```diff
@@ -56,10 +49,8 @@ public static void tearDown() throws Exception {
...
     @AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
     public void launchOpenTracing10Tck() throws Exception {
-        String bucketName = "com.ibm.ws.opentracing_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing10Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withDefaultSuiteFileName()
+    		.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.opentracing_fat_tck/fat/src/org/eclipse/microprofile/opentracing/fat/tck/OpentracingTCKLauncher.java]: Tab-space mix

File: dev/com.ibm.ws.microprofile.rest.client11_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 10]
```diff
@@ -63,11 +63,9 @@ public static void tearDown() throws Exception {
...
     	// Skip running on the windows platform when not running locally.
     	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-    		String bucketName = "com.ibm.ws.microprofile.rest.client11_fat_tck";
-    		String testName = this.getClass() + ":testRestClient11Tck";
-    		Type type = Type.MICROPROFILE;
-    		String specName = "Rest Client";
-    		TCKRunner.runTCK(server, bucketName, testName, type, specName);
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.rest.client11_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix - this one was already a mess, but could we clean it up please?

File: dev/com.ibm.ws.microprofile.rest.client12_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 11]
```diff
@@ -55,11 +55,9 @@ public static void tearDown() throws Exception {
...
     	// Skip running on the windows platform when not running locally.
     	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-    		String bucketName = "com.ibm.ws.microprofile.rest.client12_fat_tck";
-    		String testName = this.getClass() + ":testRestClient12Tck";
-    		Type type = Type.MICROPROFILE;
-    		String specName = "Rest Client";
-    		TCKRunner.runTCK(server, bucketName, testName, type, specName);
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.rest.client12_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix (previously a mess)

File: dev/com.ibm.ws.microprofile.rest.client13_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 12]
```diff
@@ -55,11 +55,9 @@ public static void tearDown() throws Exception {
...
     	// Skip running on the windows platform when not running locally.
     	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-    		String bucketName = "com.ibm.ws.microprofile.rest.client13_fat_tck";
-    		String testName = this.getClass() + ":testRestClient13Tck";
-    		Type type = Type.MICROPROFILE;
-    		String specName = "Rest Client";
-    		TCKRunner.runTCK(server, bucketName, testName, type, specName);
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.rest.client13_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix (previously a mess)

File: dev/com.ibm.ws.microprofile.rest.client14_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 13]
```diff
@@ -55,11 +55,9 @@ public static void tearDown() throws Exception {
...
     	// Skip running on the windows platform when not running locally.
     	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-    		String bucketName = "com.ibm.ws.microprofile.rest.client14_fat_tck";
-    		String testName = this.getClass() + ":testRestClient14Tck";
-    		Type type = Type.MICROPROFILE;
-    		String specName = "Rest Client";
-    		TCKRunner.runTCK(server, bucketName, testName, type, specName);
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.rest.client14_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix (previously a mess)

File: dev/com.ibm.ws.microprofile.rest.client_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 14]
```diff
@@ -54,12 +54,10 @@ public static void tearDown() throws Exception {
...
-        if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-            String bucketName = "com.ibm.ws.microprofile.rest.client_fat_tck";
-            String testName = this.getClass() + ":testRestClient10Tck";
-            Type type = Type.MICROPROFILE;
-            String specName = "Rest Client";
-            TCKRunner.runTCK(server, bucketName, testName, type, specName);
+        if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) {             
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.rest.client_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix

File: dev/io.openliberty.microprofile.graphql.2.0.internal_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java [hunk 15]
```diff
@@ -54,11 +54,9 @@ public static void tearDown() throws Exception {
...
-    public void testGraphQL20Tck() throws Exception {
-        String bucketName = "io.openliberty.microprofile.graphql.2.0.internal_fat_tck";
-        String testName = this.getClass() + ":testGraphQL20Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "GraphQL";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
+    public void testGraphQL20Tck() throws Exception {        
+        TCKRunner.build(server, Type.MICROPROFILE, "GraphQL")
+        	.withDefaultSuiteFileName()
+        	.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.graphql.2.0.internal_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java]: Tab-space mix

File: dev/io.openliberty.microprofile.health.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/health30/tck/Health30TCKLauncher.java [hunk 16]
```diff
@@ -59,11 +59,10 @@ public void launchHealth30Tck() throws Exception {
...
 
-        String bucketName = "io.openliberty.microprofile.health.3.0.internal_fat_tck";
-        String testName = this.getClass() + ":launchHealth30Tck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Health";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, additionalProps);
+        TCKRunner.build(server, Type.MICROPROFILE, "Health")
+                        .withDefaultSuiteFileName()
+                        .withAdditionalMvnProps(additionalProps)
+                        .runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.health.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/health30/tck/Health30TCKLauncher.java:65]: Copyright date not updated

File: dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingRestClientTCKLauncher.java [hunk 17]
```diff
@@ -57,10 +57,9 @@ public static void tearDown() throws Exception {
...
         String suiteName = "rest-client-tck-suite.xml";
-        String bucketName = "io.openliberty.opentracing.2.0.internal_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing20RestClientTck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+        
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+        	.withSuiteFileName(suiteName)
+        	.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingRestClientTCKLauncher.java]: Tab-space mix

File: dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncher.java [hunk 18]
```diff
@@ -30,30 +30,28 @@
...
-        TCKRunner.runTCK(server, bucketName, testName, type, specName);
-    }
+	@Test
+	@AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
+	public void launchOpenTracing20Tck() throws Exception {
+		
+		TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+			.withDefaultSuiteFileName()
+			.runTCK();
+	}
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncher.java]: Converting this whole file to tabs is certainly better than leaving it as a mix, but I'm unsure if it was deliberate as the other files in the project appear to use spaces.

File: dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncherMicroProfile.java [hunk 19]
```diff
@@ -50,11 +50,10 @@ public static void tearDown() throws Exception {
...
         String suiteName = "tck-and-rest-client-tck.xml";
-        String bucketName = "io.openliberty.opentracing.2.0.internal_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing20TckMP";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withSuiteFileName(suiteName)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.2.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncherMicroProfile.java]: Tab-space mix

File: dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingRestClientTCKLauncher.java [hunk 20]
```diff
@@ -52,10 +52,9 @@ public static void tearDown() throws Exception {
...
         String suiteName = "rest-client-tck-suite.xml";
-        String bucketName = "io.openliberty.opentracing.3.0.internal_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing30RestClientTck";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+        
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withSuiteFileName(suiteName)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingRestClientTCKLauncher.java]: Tab-space mix

File: dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncher.java [hunk 21]
```diff
@@ -30,30 +30,28 @@
...
+	}
+
+	@Test
+	@AllowedFFDC // The tested deployment exceptions cause FFDC so we have to allow for this.
+	public void launchOpenTracing30Tck() throws Exception {
+
+		TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+			.withDefaultSuiteFileName()
+			.runTCK();
+	}
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncher.java]: Is the change to all-tabs deliberate here too?

File: dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncherMicroProfile.java [hunk 22]
```diff
@@ -50,10 +50,9 @@ public static void tearDown() throws Exception {
...
         String suiteName = "tck-and-rest-client-tck.xml";
-        String bucketName = "io.openliberty.opentracing.3.0.internal_fat_tck";
-        String testName = this.getClass() + ":launchOpenTracing30TckMP";
-        Type type = Type.MICROPROFILE;
-        String specName = "Open Tracing";
-        TCKRunner.runTCK(server, bucketName, testName, type, specName, suiteName);
+        
+        TCKRunner.build(server, Type.MICROPROFILE, "Open Tracing")
+    		.withSuiteFileName(suiteName)
+    		.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.opentracing.3.0.internal_fat_tck/fat/src/io/openliberty/microprofile/opentracing/internal/tck/OpentracingTCKLauncherMicroProfile.java]: Tab-space mix

File: dev/io.openliberty.microprofile.rest.client.2.0.internal_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java [hunk 23]
```diff
@@ -71,12 +71,10 @@ public static void tearDown() throws Exception {
...
-    	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) { 
-    		String bucketName = "io.openliberty.microprofile.rest.client.2.0.internal_fat_tck";
-    		String testName = this.getClass() + ":testRestClient20Tck";
-    		Type type = Type.MICROPROFILE;
-    		String specName = "Rest Client";
-    		TCKRunner.runTCK(server, bucketName, testName, type, specName);
+    	if (!(isWindows) || FATRunner.FAT_TEST_LOCALRUN) {     		
+            TCKRunner.build(server, Type.MICROPROFILE, "Rest Client")
+            	.withDefaultSuiteFileName()
+            	.runTCK();
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.rest.client.2.0.internal_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java]: Tab-space mix (previously a mess)

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java [hunk 24]
```diff
@@ -159,4 +175,104 @@ public String getSHA1() {
...
+     * Returns a human readable full specification name such as "Jakarta Data 1.0"
+     *
+     * @return
+     */
+    public String getFullSpecName() {
+        switch (type) {
+            case JAKARTA:
+                return "Jakarta " + specName + " " + getSpecVersion();
+            case MICROPROFILE:
+                return "Microprofile " + specName + " " + getSpecVersion();
```
Comment (Azquelt) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java]:
```suggestion
                return "MicroProfile " + specName + " " + getSpecVersion();
```

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java [hunk 25]
```diff
@@ -159,4 +175,104 @@ public String getSHA1() {
...
+            default:
+                return "UKNOWN";
```
Comment (Azquelt) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java]:
```suggestion
                return "UNKNOWN";
```

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java [hunk 26]
```diff
@@ -159,4 +175,104 @@ public String getSHA1() {
...
+
+        }
+    }
+
+    String getSpecNameForURL() {
+        return specName.toLowerCase().replace(" ", "-");
+    }
```
Comment (Azquelt) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java]: Should this method be either public or private?

File: dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java [hunk 27]
```diff
@@ -159,4 +175,104 @@ public String getSHA1() {
...
+        filename = filename.replace(",", "-");
+        filename = filename.replace("_", "-");
+        filename = filename.replace(":", "-");
+        filename = filename.replace(";", "-");
+
+        return filename;
+    }
+
+    public String getReadableRepeatName() {
+        String redeableRepeatName = getRepeat();
```
Comment (Azquelt) [inline dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsInfo.java]: `redeableRepeatName` -> `readableRepeatName`

**Review by Azquelt — CHANGES_REQUESTED** [review]
This looks like an excellent improvement to the usability of `TCKRunner`, thank you!

My only comments are small formatting issues (likely caused by older projects without good project formatting settings) and a few typos.

**Comment by KyleAure** [comment]
@Azquelt - Thanks for the thorough review.  I suffered through and updated my eclipse and re-built everything to get the formatting issues resolved. 🎊

**Comment by KyleAure** [comment]
#build - personal build
#libby - re-check headers

**Comment by KyleAure** [comment]
#build - redo personal build after rebase and fix
#libby - recheck headers

**Review by Azquelt — APPROVED** [review]
Looks great, thanks :+1:

**Review by jakub-pomykala — APPROVED** [review]
LGTM!

**Comment by KyleAure** [comment]
Pipeline: passed
RTC Build: failed with one known defect

ready to merge once build con improves.

**Comment by KyleAure** [comment]
Double checked to ensure no new TCK projects were pushed to Integration prior to merging this PR. 
🤞🏼 This does not end up causing a soft merge conflict.

**Pull request merged**